config/config.md

```diff
@@ -335,6 +335,8 @@
 | `init_regions_in_background` | Bool | `false` | Initialize all regions in the background during the startup.<br/>By default, it provides services after all regions have been initialized. |
 | `enable_telemetry` | Bool | `true` | Enable telemetry to collect anonymous usage data. |
 | `init_regions_parallelism` | Integer | `16` | Parallelism of initializing regions. |
+| `max_concurrent_queries` | Integer | `0` | The maximum current queries allowed to be executed. Zero means unlimited. |
+| `concurrent_query_limiter_timeout` | String | `25ms` | The timeout of acquiring concurrency permits. |
```

Comment:
Do we need this option? Despite the implementation details (`acquire` vs. `try_acquire`), it seems like users would hardly adjust this option. And exposing such an option would also limit us from implementing a better behavior in the future, like "only pending 30 awaiting requests in the queue, for at most 100ms".

We can hide such logic as our system's internal details. WDYT @sunng87 @evenyag

Reply:
I agree remove this option will give better flexibility for future. I think `try_acquire` is a special case that is equivalent to 0 timeout i would prefer more flexibility in this case. I removed the `concurrent_query_limiter_timeout` in DataNode options and hardcoded 25ms wait with a TODO. In future versions i think we could update the `max_concurrent_queries` and `concurrent_query_limiter_timeout` dynamically via sql such as `SET max_concurrent_queries = 8;`